code/datums/status_effects/neutral.dm

```diff
@@ -135,3 +135,126 @@
+	var/original_alpha = 255
+	///All incoming brute/burn/stamina damage on the protector is multiplied by this, whether it was directed at the protectee or not
+	var/defensive_buff_factor = 0.75
+	///Paralyze the protectee for this long so they don't accidentally slip away prematurely
+	var/paralyze_duration = 1.5 SECONDS
+
+/obj/screen/alert/status_effect/protected
+	name = "Protected"
+	desc = "Someone is covering and leading you while protecting you from attacks. You can click here to make them release you."
+	icon_state = "aimed"
+
+/obj/screen/alert/status_effect/protector
+	name = "Protecting"
+	desc = "You're currently leading and defending someone from attacks, while benefiting from a defensive buff. You can click here to manually release them."
+	icon_state = "aimed"
+
+/obj/screen/alert/status_effect/protected/Click()
+	var/mob/living/L = usr
+	if(L != owner)
+		return
+	QDEL_NULL(attached_effect)
+	qdel(src)
+
+/obj/screen/alert/status_effect/protector/Click()
+	var/mob/living/L = usr
+	if(L != owner)
+		return
+	QDEL_NULL(attached_effect)
+	qdel(src)
+
+/datum/status_effect/aegis/on_creation(mob/living/new_owner, mob/living/protector)
+	src.protector = protector
+	linked_alert_protector = protector.throw_alert(id, alert_type_protector)
+	linked_alert_protector.attached_effect = src
+	RegisterSignal(new_owner, COMSIG_CARBON_ATTACKED_BY, .proc/interrupt_hit)
+	RegisterSignal(new_owner, COMSIG_MOVABLE_MOVED, .proc/check_protectee_dist)
+	RegisterSignal(new_owner, list(COMSIG_PROJECTILE_PREHIT, COMSIG_PROJECTILE_POINT_BLANK), .proc/interrupt_bullet)
+	RegisterSignal(protector, COMSIG_MOVABLE_MOVED, .proc/take_me_with_you)
+	RegisterSignal(protector, SIGNAL_ADDTRAIT(TRAIT_INCAPACITATED), .proc/cancel_protection)
+	..()
+
+/datum/status_effect/aegis/on_apply()
+	owner.visible_message("<span class='notice'><b>[protector] covers [owner] with [protector.p_their()] body, guarding against attacks!</b></span>", "<span class='notice'><b>[protector] covers you with [protector.p_their()] body, guarding against attacks!</b></span>")
+	examine_text = "<span class='notice'><b>[owner.p_they(TRUE)] [owner.p_are()] being guarded closely by [protector].</b></span>"
+	original_alpha = owner.alpha
+	owner.alpha = 100
+	playsound(owner, 'sound/weapons/fwoosh.ogg', 75, FALSE)
+	protector.forceMove(owner.loc)
+	owner.SetKnockdown(duration)
+	owner.SetImmobilized(paralyze_duration)
+	apply_defensive_buff()
+	return ..()
+
+/datum/status_effect/aegis/on_remove()
+	if(owner && protector)
+		owner.visible_message("<span class='notice'>[protector] ceases guarding [owner].</span>", "<span class='notice'><b>[protector] ceases guarding you.</b></span>")
+
+	if(owner)
+		owner.alpha = original_alpha
+		UnregisterSignal(owner, list(COMSIG_CARBON_ATTACKED_BY, COMSIG_MOVABLE_MOVED, COMSIG_PROJECTILE_PREHIT, COMSIG_PROJECTILE_POINT_BLANK))
+		owner.SetKnockdown(0)
```

Comment:
so I can end this early to clear the knockdown from a baton, or the hardstun from a flash?

hm...

Reply:
?? no? hardstun is paralyze not knockdown, they're different effects.

Reply:
it clears the hardstun right underneath this line, no?